Automattic/wp-calypso
# composite-checkout: Remove @wordpress/data

#### Changes proposed in this Pull Request

The `@wordpress/data` package is included in the `@automattic/composite-checkout` package as a convenience for its children, re-exporting many of that packages' methods and creating a default registry. However, that's a big price to pay for this convenience since `@wordpress/data` includes `lodash`. Previously the package's built-in payment methods used the registry so this made sense, but now that the package has no built-in payment methods, it makes sense to remove this dependency and allow the consumer to decide how to implement its data layer.

This is also related to [creating](https://github.com/Automattic/wp-calypso/pull/52890) a mini-cart to use in wp-admin, since the (unused but not tree-shakable) `@wordpress/data` dependency adds quite a lot of overhead to the mini-cart app.

This PR removes the `@wordpress/data` dependency from that package. In so doing, it removes the following pieces of the package's public API, most of which can be found in `@wordpress/data` directly:

- `registerStore` (migrated)
- `useDispatch` (migrated)
- `useSelect` (migrated)
- `createRegistry` (could be migrated but no longer used by anything)
- `useRegisterStore` (not available any more)
- `useRegistry` (not available any more)
- `defaultRegistry` (not available any more)

As long as we're touching all uses of the store, this PR also renames the `@wordpress/data` store used by checkout from `'wpcom'` to `'wpcom-checkout'`. Hopefully this will prevent confusion and misuse by future developers since the previous name could refer to anything at all.

#### To do

- [x] Replace the registry in the package tests with some other local state.
- [x] Replace the registry in the demo with some other local state.
- [x] Use default wordpress/data registry everywhere a registry is needed.
- [x] Replace any instances of the removed API functions.
- [x] Figure out why the demo no longer does anything (same in trunk; I think it's [this bug](https://github.com/storybookjs/storybook/issues/12881)).

#### Testing instructions

Testing this is fairly simple because the registry is so essential to checkout that a bug likely will crash calypso or prevent it from building.

- Add a product to your cart and visit checkout.
- Complete the checkout form and submit it.
- Verify that there are no fatal errors.
